p256/src/arithmetic/hash2curve.rs

```diff
@@ -0,0 +1,283 @@
+            0xffff_ffff_ffff_fffeu64,
+            0x0000_0002_0000_0000u64,
+            0x0000_0000_0000_0003u64,
+        ]);
+        let d0 = FieldElement([
+            u64::from_be_bytes(data[16..24].try_into().unwrap()),
+            u64::from_be_bytes(data[8..16].try_into().unwrap()),
+            u64::from_be_bytes(data[0..8].try_into().unwrap()),
+            0,
+        ])
+        .to_montgomery();
+        let d1 = FieldElement([
+            u64::from_be_bytes(data[40..48].try_into().unwrap()),
+            u64::from_be_bytes(data[32..40].try_into().unwrap()),
+            u64::from_be_bytes(data[24..32].try_into().unwrap()),
+            0,
+        ])
+        .to_montgomery();
+        d0 * F_2_192 + d1
+    }
+}
+
+impl Sgn0 for FieldElement {
+    fn sgn0(&self) -> Choice {
+        FieldElement::sgn0(self)
+    }
+}
+
+impl OsswuMap for FieldElement {
+    const PARAMS: OsswuMapParams<Self> = OsswuMapParams {
+        c1: [
+            0xffff_ffff_ffff_ffff,
+            0x0000_0000_3fff_ffff,
+            0x4000_0000_0000_0000,
+            0x3fff_ffff_c000_0000,
+        ],
+        c2: FieldElement([
+            0x53e4_3951_f64f_dbe7,
+            0xb280_6c63_966a_1a66,
+            0x1ac5_d59c_3298_bf50,
+            0xa332_3851_ba99_7e27,
+        ]),
+        map_a: FieldElement([
+            0xffff_ffff_ffff_fffc,
+            0x0000_0003_ffff_ffff,
+            0x0000_0000_0000_0000,
+            0xffff_fffc_0000_0004,
+        ]),
+        map_b: FieldElement([
+            0xd89c_df62_29c4_bddf,
+            0xacf0_05cd_7884_3090,
+            0xe5a2_20ab_f721_2ed6,
+            0xdc30_061d_0487_4834,
+        ]),
+        z: FieldElement([
+            0xffff_ffff_ffff_fff5,
+            0x0000_000a_ffff_ffff,
+            0x0000_0000_0000_0000,
+            0xffff_fff5_0000_000b,
+        ]),
+    };
+}
+
+impl MapToCurve for FieldElement {
+    type Output = ProjectivePoint;
+
+    fn map_to_curve(&self) -> Self::Output {
+        let (qx, qy) = self.osswu();
+
+        AffinePoint {
+            x: qx,
+            y: qy,
+            infinity: Choice::from(0u8),
+        }
+        .into()
+    }
+}
+
+impl CofactorGroup for ProjectivePoint {
+    type Subgroup = ProjectivePoint;
+
+    fn clear_cofactor(&self) -> Self::Subgroup {
+        *self
+    }
+
+    fn into_subgroup(self) -> CtOption<Self::Subgroup> {
+        CtOption::new(self, 1.into())
+    }
+
+    fn is_torsion_free(&self) -> Choice {
+        1.into()
+    }
+}
+
+impl FromOkm for Scalar {
+    type Length = U48;
+
+    fn from_okm(data: &GenericArray<u8, Self::Length>) -> Self {
+        let n = NonZero::new(U384::from_be_hex(
+            "00000000000000000000000000000000ffffffff00000000ffffffffffffffffbce6faada7179e84f3b9cac2fc632551",
+        ))
+        .unwrap();
+
+        let bytes = U256::from_be_slice(&(U384::from_be_slice(data) % n).to_be_byte_array()[16..]);
+        Scalar(bytes)
+    }
+}
```

Comment:
I can't do the magic that @mikelodder7 did, so this is my amateur solution. It works.

Reply:
I was taught some black magic by @mikelodder7 in the meantime.